Source/WebCore/Modules/webauthn/fido/DeviceRequestConverter.cpp

```diff
@@ -241,22 +259,36 @@ Vector<uint8_t> encodeGetAssertionRequestAsCBOR(const Vector<uint8_t>& hash, con
         cborMap[CBORValue(3)] = CBORValue(WTFMove(allowListArray));
     }
 
-    if (authenticatorSupportedExtensions.size() && options.extensions) {
+    if ((authenticatorSupportedExtensions.size() && options.extensions) || hmacSecretParams) {
         CBORValue::MapValue extensionsMap;
-        auto largeBlobInputs = options.extensions->largeBlob;
-        if (largeBlobInputs && authenticatorSupportedExtensions.contains("largeBlob"_s)) {
-            CBORValue::MapValue largeBlobMap;
 
-            if (largeBlobInputs->read)
-                largeBlobMap[CBORValue("read"_s)] = CBORValue(largeBlobInputs->read.value());
+        if (options.extensions) {
+            auto largeBlobInputs = options.extensions->largeBlob;
+            if (largeBlobInputs && authenticatorSupportedExtensions.contains("largeBlob"_s)) {
+                CBORValue::MapValue largeBlobMap;
 
-            if (largeBlobInputs->write)
-                largeBlobMap[CBORValue("write"_s)] = CBORValue(BufferSource { WTFMove(*largeBlobInputs->write) });
+                if (largeBlobInputs->read)
+                    largeBlobMap[CBORValue("read"_s)] = CBORValue(largeBlobInputs->read.value());
+
+                if (largeBlobInputs->write)
+                    largeBlobMap[CBORValue("write"_s)] = CBORValue(BufferSource { WTFMove(*largeBlobInputs->write) });
+
+                extensionsMap[CBORValue("largeBlob"_s)] = CBORValue(WTFMove(largeBlobMap));
+            }
+        }
 
-            extensionsMap[CBORValue("largeBlob"_s)] = CBORValue(WTFMove(largeBlobMap));
+        // Add hmac-secret extension with encrypted salt parameters
+        if (hmacSecretParams && authenticatorSupportedExtensions.contains(kExtensionHmacSecret)) {
+            CBORValue::MapValue hmacSecretMap;
+            hmacSecretMap[CBORValue(kHmacSecretKeyAgreementKey)] = CBORValue(WTFMove(hmacSecretParams->keyAgreement));
+            hmacSecretMap[CBORValue(kHmacSecretSaltEncKey)] = CBORValue(WTFMove(hmacSecretParams->saltEnc));
+            hmacSecretMap[CBORValue(kHmacSecretSaltAuthKey)] = CBORValue(WTFMove(hmacSecretParams->saltAuth));
+            hmacSecretMap[CBORValue(kHmacSecretPinUvAuthProtocolKey)] = CBORValue(hmacSecretParams->protocol);
+            extensionsMap[CBORValue(kExtensionHmacSecret)] = CBORValue(WTFMove(hmacSecretMap));
         }
 
-        cborMap[CBORValue(4)] = CBORValue(WTFMove(extensionsMap));
+        if (!extensionsMap.empty())
+            cborMap[CBORValue(4)] = CBORValue(WTFMove(extensionsMap));
```

Comment:
This magic number should probably be a `kExtensionsMap` constant (or whatever the right name would be).